pandas/tests/series/test_operators.py

```diff
@@ -537,6 +537,23 @@ def test_comparison_flex_alignment_fill(self):
         exp = pd.Series([True, True, False, False], index=list('abcd'))
         assert_series_equal(left.gt(right, fill_value=0), exp)
 
+    def test_comparison_with_index(self):
+        # GH22092
+        ser = Series([True, True, False, False])
+        idx = Index([True, False, True, False])
+
+        expected = Series([True, False, False, False])
+        result = ser & idx
```

Comment:
> I think we DO allow this on index types which don't makes sense, but we DO raise on Series,

Yah I've opened a couple of Issues about this.  The existing Series ops don't handle dt64/td64 at all, and are not super-clear for some other dtypes.  The reversed `op(Index, Series)` ops are a separate can of worms because those are set operations.

I don't think this PR fixes [4] (yet), needs to chance ABCIndex to ABCIndexClass (like #22173)

Reply:
ok this is fine then